verification/verification.go

```diff
@@ -46,6 +46,88 @@ func pad(plaintext []byte, bsize int) ([]byte, error) {
 	return plaintext, nil
 }
 
+// GenerateChallengeEx takes the EK (rsa.PublicKey), the public half of
+// the AIK to be challenged and a secret. It then symmetrically encrypts the
+// secret with a randomly generated AES key and Asymmetrically encrypts the
+// AES key with the public half of the EK. These can then be provided to the
+// TPM in order to ensure that the AIK is under the control of the TPM. It
+// returns the asymmetrically and symmetrically encrypted data, along with
+// any error.
+func GenerateChallengeEx(pubkey *rsa.PublicKey, aikpub []byte, secret []byte) (asymenc []byte, symenc []byte, err error) {
```

Comment:
Can you refactor the shared code between the two methods into an unexported method or helper?

Reply:
done